File: managers/src/main/java/com/github/managers/services/impl/SpecificationService.java
package com.github.managers.services.impl;

import com.github.managers.dto.SpecificationDto;
import com.github.managers.services.ISpecificationService;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
public class SpecificationService implements ISpecificationService {

    @Override
    public Optional<SpecificationDto> create(Long productId, SpecificationDto s) {
        return Optional.empty();
    }

    @Override
    public Optional<SpecificationDto> readById(Long id) {
        return Optional.empty();
    }

    @Override
    public void update(SpecificationDto s) {

    }
}

File: server-mocks/src/main/kotlin/com/github/server/mocks/services/impl/SpecificationService.kt
package com.github.server.mocks.services.impl

import com.github.server.mocks.payload.Specification
import com.github.server.mocks.services.ISpecificationService
import com.github.server.mocks.utils.IdsGenerator
import org.springframework.stereotype.Service

@Service
class SpecificationService: ISpecificationService {

    var mock: Specification = Specification(
            1,
            "Phone",
            "This is phone spec."
    );

    override fun create(specification: Specification): Specification {
        return mock
    }

    override fun readById(id: Long): Specification {
        return mock
    }

    override fun update(specification: Specification) {
    }

}